src/vaadin-grid-filter-mixin.js

```diff
@@ -29,13 +29,38 @@ export const FilterMixin = (superClass) =>
 
     /** @private */
     _filterChanged(e) {
-      if (this._filters.indexOf(e.target) === -1) {
-        this._filters.push(e.target);
+      e.stopPropagation();
+
+      this.__updateFilter(e.target);
+      this.__applyFilters();
+    }
+
+    __removeFilters(filters) {
+      if (filters.length == 0) {
+        return;
       }
 
-      e.stopPropagation();
+      this._filters.forEach((filter) => {
+        const filterIndex = this._filters.indexOf(filter);
+        if (filterIndex !== -1) {
+          this._filters.splice(filterIndex, 1);
+        }
+      });
+      this.__applyFilters();
+    }
+
+    /** @private */
+    __updateFilter(filter) {
```

Comment:
`__updateFilter` sounds confusing, since it only adds. What is the intent here?

Reply:
Renamed to `__addFilter`